Examples/Algorithms/Fitting/include/ActsExamples/Fitting/DirectedFittingAlgorithm.hpp

```diff
@@ -0,0 +1,73 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/TrackFitting/KalmanFitter.hpp"
+#include "ActsExamples/EventData/SimSourceLink.hpp"
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/Framework/BareAlgorithm.hpp"
+#include "ActsExamples/Plugins/BField/BFieldOptions.hpp"
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace ActsExamples {
+
+class DirectedFittingAlgorithm final : public BareAlgorithm {
+ public:
+  using FitterResult = Acts::Result<Acts::KalmanFitterResult<SimSourceLink>>;
+  /// Fit function that takes input measurements, initial trackstate and fitter
+  /// options and returns some fit-specific result.
+  using FitterFunction = std::function<FitterResult(
+      const std::vector<SimSourceLink>&, const TrackParameters&,
+      const Acts::KalmanFitterOptions<Acts::VoidOutlierFinder>&,
+      const std::vector<const Acts::Surface*>&)>;
+
+  /// Create the fitter function implementation.
+  ///
+  /// The magnetic field is intentionally given by-value since the variant
+  /// contains shared_ptr anyways.
+  static FitterFunction makeFitterFunction(
+      Options::BFieldVariant magneticField);
+
+  struct Config {
+    /// Input source links collection.
+    std::string inputSourceLinks;
+    /// Input proto tracks collection, i.e. groups of hit indices.
+    std::string inputProtoTracks;
+    /// Input initial track parameter estimates for for each proto track.
+    std::string inputInitialTrackParameters;
+    /// Input surface sequence for DirectNavigator
+    std::string inputSurfaceSequence;
```

Comment:
The `inputSurfaceSequence` is track-dependent, so should be derived based on the input source links for a track in some way.

Reply:
This is a good point, since the source links for a given track essentially define what the `inputSurfaceSequence` will be.